cmd/kubeadm/app/apis/output/v1alpha3/types.go

```diff
@@ -0,0 +1,126 @@
+	CurrentVersion string `json:"currentVersion"`
+	NewVersion     string `json:"newVersion"`
+}
+
+// ComponentConfigVersionState describes the current and desired version of a component config
+type ComponentConfigVersionState struct {
+	// Group points to the Kubernetes API group that covers the config
+	Group string `json:"group"`
+
+	// CurrentVersion is the currently active component config version
+	// NOTE: This can be empty in case the config was not found on the cluster or it was unsupported
+	// kubeadm generated version
+	CurrentVersion string `json:"currentVersion"`
+
+	// PreferredVersion is the component config version that is currently preferred by kubeadm for use.
+	// NOTE: As of today, this is the only version supported by kubeadm.
+	PreferredVersion string `json:"preferredVersion"`
+
+	// ManualUpgradeRequired indicates if users need to manually upgrade their component config versions. This happens if
+	// the CurrentVersion of the config is user supplied (or modified) and no longer supported. Users should upgrade
+	// their component configs to PreferredVersion or any other supported component config version.
+	ManualUpgradeRequired bool `json:"manualUpgradeRequired"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// UpgradePlan represents information about upgrade plan for the output
+// produced by 'kubeadm upgrade plan'
+type UpgradePlan struct {
+	metav1.TypeMeta
+
+	Components []ComponentUpgradePlan `json:"components"`
+
+	ConfigVersions []ComponentConfigVersionState `json:"configVersions"`
+}
+
+// Certificate represents information about check-expiration for one certificate
+type Certificate struct {
+	// Name of the certificate
+	Name string `json:"name"`
+	// ExpirationDate defines certificate expiration date.
+	ExpirationDate metav1.Time `json:"expirationDate"`
+	// ResidualTime represents the time missing to expiration
+	ResidualTime time.Duration `json:"residualTime"`
+	// ExternallyManaged defines if the certificate is externally managed
+	IsExternallyManaged bool `json:"isExternallyManaged"`
+	// CAName represents the name of the CA that signed the certificate
+	CAName string `json:"caName"`
+	// IsMissing represents if the certificate is missing
+	IsMissing bool `json:"isMissing"`
+}
+
+// CertificateAuthority represents information about check-expiration for one certificate authority
+type CertificateAuthority struct {
+	// Name of the certificate authority
+	Name string `json:"name"`
+	// ExpirationDate defines certificate authority expiration date.
+	ExpirationDate metav1.Time `json:"expirationDate"`
+	// ResidualTime represents the time missing to expiration
+	ResidualTime time.Duration `json:"residualTime"`
+	// ExternallyManaged defines if the certificate authority is externally managed
+	IsExternallyManaged bool `json:"isExternallyManaged"`
+	// IsMissing represents if the certificate authority is missing
+	IsMissing bool `json:"isMissing"`
```

Comment:
```suggestion
	ExternallyManaged bool `json:"externallyManaged"`
	// IsMissing represents if the certificate authority is missing
	Missing bool `json:"missing"`
```

Not quite sure about the naming thing and `missing: true` vs `isMissing: true`. I prefer without-is.